system/Database/MySQLi/Result.php

```diff
@@ -87,16 +87,52 @@ public function getFieldNames(): array
 	 */
 	public function getFieldData(): array
 	{
+		static $data_types = [
+			MYSQLI_TYPE_DECIMAL		=> 'decimal',
+			MYSQLI_TYPE_NEWDECIMAL	=> 'newdecimal',
+			MYSQLI_TYPE_FLOAT		=> 'float',
+			MYSQLI_TYPE_DOUBLE		=> 'double',
+			
+			MYSQLI_TYPE_BIT			=> 'bit',
+			MYSQLI_TYPE_TINY		=> 'tiny',
+			MYSQLI_TYPE_SHORT		=> 'short',
+			MYSQLI_TYPE_LONG		=> 'long',
+			MYSQLI_TYPE_LONGLONG	=> 'longlong',
+			MYSQLI_TYPE_INT24		=> 'int24',
+			
+			MYSQLI_TYPE_YEAR		=> 'year',
+			
+			MYSQLI_TYPE_TIMESTAMP	=> 'timestamp',
+			MYSQLI_TYPE_DATE		=> 'date',
+			MYSQLI_TYPE_TIME		=> 'time',
+			MYSQLI_TYPE_DATETIME	=> 'datetime',
+			MYSQLI_TYPE_NEWDATE		=> 'newdate',
+			
+			MYSQLI_TYPE_INTERVAL	=> 'interval',
+			MYSQLI_TYPE_SET			=> 'set',
+			MYSQLI_TYPE_ENUM		=> 'enum',
+			
+			MYSQLI_TYPE_VAR_STRING	=> 'var_string',
+			MYSQLI_TYPE_STRING		=> 'string',
+			MYSQLI_TYPE_CHAR		=> 'char',
+			
+			MYSQLI_TYPE_GEOMETRY	=> 'geometry',
+			MYSQLI_TYPE_TINY_BLOB	=> 'tiny_blob',
+			MYSQLI_TYPE_MEDIUM_BLOB	=> 'medium_blob',
+			MYSQLI_TYPE_LONG_BLOB	=> 'long_blob',
+			MYSQLI_TYPE_BLOB		=> 'blob',
+		];
```

Comment:
It will make a maintenance burden in the future when a new type added/removed.

Reply:
The same was done with SQL Lite driver, that why I followed the same logic. 
Cant think of a alternative for if the idea is the data returned by fielddata should be standard but blind to what database is in the background then something like a map will be needed as php dont seem to give alternative functions for it.

Reply:
I can change the line to have this logic.

$retVal[$i]->type        = isset($data_types[$data->type]) ? $data_types[$data->type] : $data->type;

then if their are future unmapped types, it wont break framework, but return the numeric value and one can then later on with update add the new type. 

New database field types is not added easily thus it wont be something that will need to be maintained for every single new mysql version I believe